### PR TITLE
Feedback

### LINKED ISSUE #2: Feedback 1
1. Tạo folder `constant/...` để lưu các giá trị define như là: conditionToWinForOrthogonal, conditionToWinForOrthogonal, minGridSize, maxGridSize

2. Có thể tạo folder `scss hoăc css`: sử dụng @apply trong tailwind để có thể sử dụng lại css chung. 
2.1 Khi viết css, nên chia ra theo thứ tự quan trọng giảm dần để dễ theo dõi: https://9elements.com/css-rule-order/

4. Sử dụng Arrow Function ES6

5. Config thêm Absolute Import: https://nextjs.org/docs/app/building-your-application/configuring/absolute-imports-and-module-aliases

### PR DESCRIPTION
**Pull request by tamy-le**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [X] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
PR for feedback from frontend senior dev to refactor code:
- [x] Create folder constants and move constants variables
- [x] Use @apply of tailwind for cleaner className
- [x] Arrow function for layout and page
- [x] Config absolute import to remove ../../ or ../ 
- [x] Using classnames to simplify and cleaner code for css class name 

Bug fixed:
- [x] Input can not type
- [x] background not complete black
- [x] warning of non initial value of state
- [x] warning of nan value   

## QA Instructions, Screenshots, Recordings

Run `npm install` and `npm run dev` then look for `localhost:3000` page

## Added/updated tests?

- [x] Yes: manually 
- [ ] No, and this is why: _
- [ ] I need help with writing tests

### CODE REVIEW

File: tic-tac-toe/src/components/Square.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
 const Square = ({ value, isWonSquare, handleClicked }) => {
-  const backgroundColor = isWonSquare ? "bg-rose-300" : "bg-white";
-  const textColor = isWonSquare ? "text-red-600	" : "text-black";
-  const fontWeight = isWonSquare ? "font-semibold" : "font-light";
-  const className = `border-4 border-indigo-500/100 float-left size-16 box-border ${fontWeight} ${textColor} text-center ${backgroundColor}`;
+	const backgroundColor = isWonSquare ? "bg-rose-300" : "bg-white";
+	const textColor = isWonSquare ? "text-red-600	" : "text-black";
```
Comment (thubuidev) [inline tic-tac-toe/src/components/Square.js]: Sao phải có space chỗ này dị "text-red-600	"

Comment (thubuidev) [reply tic-tac-toe/src/components/Square.js]: Có thể tìm hiểu thêm classNames nha: https://www.npmjs.com/package/classnames

File: tic-tac-toe/src/utils/WinnerCheck.js [hunk 2]
```diff
@@ -1,167 +1,163 @@
...
+	}
 
-  return null;
+	return null;
 };
 
 const checkDraw = (currentBoard) => {
-  return !currentBoard.includes("");
+	return !currentBoard.includes("");
 };
```
Comment (thubuidev) [inline tic-tac-toe/src/utils/WinnerCheck.js]: const checkDraw = (currentBoard) => !currentBoard.includes("");

File: tic-tac-toe/src/app/page.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
 "use client";
-import Game from "./../components/Game";
-export default function Page() {
-  return <Game />;
-}
+import Game from "components/Game";
+
+const Page = () => {
+	return <Game />;
+};
```
Comment (thubuidev) [inline tic-tac-toe/src/app/page.js]: `const Page = () => <Game />`

File: tic-tac-toe/src/components/Square.js (hunk 1 above)
Comment (tamy-le) [reply tic-tac-toe/src/components/Square.js]: > Sao phải có space chỗ này dị "text-red-600 "
:)) typooooo

File: tic-tac-toe/src/components/Game.js [hunk 4]
```diff
@@ -1,132 +1,136 @@
...
+			newSize >= minGridSize &&
+			newSize <= maxGridSize
+		) {
+			setGridSize(newSize);
+			setHistory([{ squares: Array(newSize ** 2).fill(""), index: -1 }]);
+			setStatus("Game Start");
+			setWinner("");
+			setCurrentMove(0);
+			setWinningSquares();
+			setToggle(false);
```
Comment (thubuidev) [inline tic-tac-toe/src/components/Game.js]: Tách hàm reset

File: tic-tac-toe/tailwind.config.js [hunk 5]
```diff
@@ -1,18 +1,19 @@
...
+	theme: {
+		extend: {
+			backgroundImage: {
+				"gradient-radial": "radial-gradient(var(--tw-gradient-stops))",
+				"gradient-conic":
+					"conic-gradient(from 180deg at 50% 50%, var(--tw-gradient-stops))",
+			},
+		},
+	},
+	plugins: [],
```
Comment (thubuidev) [inline tic-tac-toe/tailwind.config.js:18]: Không có cmt gì hết

**Review by thubuidev — APPROVED** [review]

**Review by tuority — APPROVED** [review]

**Pull request merged**